orttraining/orttraining/test/python/orttraining_test_pipeline_parallel_tests.py

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env python3
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+import sys
+import argparse
+
+from _test_commons import run_subprocess
+
+import logging
+
+logging.basicConfig(
+    format="%(asctime)s %(name)s [%(levelname)s] - %(message)s",
+    level=logging.DEBUG)
+log = logging.getLogger("Build")
+
+
+def parse_arguments():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--cwd", help="cwd")
+    return parser.parse_args()
+
+
+def main():
+    import torch
+    ngpus = torch.cuda.device_count()
+
+    # TODO: currently the CI machine only has 4 GPUs for parallel tests.
+    # Fill in more pipeline partition options when the machine has different GPUs counts.
+    if ngpus != 4:
+        return 0
+
+    log.info("Running distributed computation tests.")
+
+    command = ['mpirun',  '-n', '4', 'python', './orttraining_test_parallel_train_simple_model.py',
```

Comment:
You want to use ```sys.executable``` here instead of ```python```. Also, if possible, try not to hardcode the number of processes here to 4. Use what is available as the number of gpus in ```torch.cuda.device_count()```.

Reply:
It requires at right 4 GPUs and I change the condition to
```
    # Test scripts for parallel tests.
    distributed_test_files = ['./orttraining_test_parallel_train_simple_model.py']
    # parallel_test_process_number[i] is the number of processes needed to run distributed_test_files[i].
    distributed_test_process_counts = [4]

    log.info("Running parallel training tests.")
    for test_file, process_count in zip(distributed_test_files, distributed_test_process_numbers):
        if ngpus < process_count:
            log.info("SKIP: " test_file)
            continue
        log.debug('RUN: ' + test_file)
        _distributed_run(test_file, None, None, process_count)
```
Because DxHxP parallel tests use their own specific numbers of GPUs, we can't use torch.cuda.device_count().

---
In reply to: [549429868](https://github.com/microsoft/onnxruntime/pull/5815#discussion_r549429868) [](ancestors = 549429868)

Reply:
If there are more files in the future that all relate to the same test suite, may we create a folder with those test files and call the test files from the main entry point orttraining/orttraining/test/python/orttraining_test_pipeline_parallel_tests.py which can be outside this folder. Basically:
```
- test
  - python
    - orttraining_test_pipeline_parallel_tests.py
    - pipeline_parallel_tests
      - orttraining_test_parallel_train_simple_model.py
      - other_future_tests.py
```

This is similar to the checkpoint tests defined in the [main entry point file](https://github.com/microsoft/onnxruntime/blob/master/orttraining/orttraining/test/python/orttraining_test_checkpoint.py) and the [test folder](https://github.com/microsoft/onnxruntime/tree/master/orttraining/orttraining/test/python/checkpoint).

Reply:
Done. We now have
```
- test
  - python
    - orttraining_test_dhp_parallel_tests.py
    - dhp_pipeline
      - orttraining_test_parallel_train_simple_model.py
      - other_future_tests.py
```